django/db/models/functions/window.py

```diff
@@ -0,0 +1,120 @@
+from django.db.models import Aggregate, FloatField, IntegerField
+
+__all__ = [
+    'RowNumber', 'Rank', 'DenseRank', 'Ntile', 'FirstValue',
+    'PercentRank', 'NthValue', 'LastValue', 'CumeDist', 'Lag',
+    'Lead',
+]
+
+
+class CumeDist(Aggregate):
```

Comment:
I decided to derive them all from `Func` because they're not aggregate functions in the same as `Count` and `Sum` which don't need to be part of an expression such as `WITHIN GROUP` or `OVER`. The `GROUP BY` required by the aggregate functions is also not needed in many cases, so I think your initial verdict was right.